pmd-java/src/main/java/net/sourceforge/pmd/lang/java/ast/Comment.java

```diff
@@ -5,45 +5,102 @@
-        }
     }
 
     @Override
     public String toString() {
         return getImage();
     }
 
-    private void findJavadocs(String commentText) {
+    /**
+     * Filters the comment by removing the leading comment marker (like {@code *}) of each line
+     * as well as the start markers ({@code //}, {@code /*} or {@code /**}
+     * and the end markers (<code>&#x2a;/</code>).
+     * Also leading and trailing empty lines are removed.
+     *
+     * @return the filtered comment
+     */
+    public String getFilteredComment() {
+        List<String> lines = multiLinesIn();
+        lines = trim(lines);
+        return StringUtils.join(lines, PMD.EOL);
+    }
 
-        Collection<JavadocElement> kids = new ArrayList<>();
+    /**
+     * Removes the leading comment marker (like {@code *}) of each line
+     * of the comment as well as the start marker ({@code //}, {@code /*} or {@code /**}
+     * and the end markers (<code>&#x2a;/</code>).
+     *
+     * @param comment the raw comment
+     * @return List of lines of the comments
+     */
+    private List<String> multiLinesIn() {
+        String[] lines = getImage().split("\\R");
+        List<String> filteredLines = new ArrayList<>(lines.length);
 
-        Map<String, Integer> tags = CommentUtil.javadocTagsIn(commentText);
-        for (Map.Entry<String, Integer> entry : tags.entrySet()) {
-            JavadocTag tag = JavadocTag.tagFor(entry.getKey());
-            if (tag == null) {
-                continue;
+        for (String rawLine : lines) {
+            String line = rawLine.trim();
+
+            Matcher allMatcher = COMMENT_LINE_COMBINED.matcher(line);
+            if (allMatcher.matches()) {
+                filteredLines.add(allMatcher.group(1).trim());
             }
-            kids.add(new JavadocElement(getBeginLine(), getBeginLine(),
-                    // TODO valid?
-                    entry.getValue() + 1, entry.getValue() + tag.label.length() + 1, tag));
         }
 
-        children = kids.toArray(new Node[0]);
+        return filteredLines;
     }
 
+    /**
+     * Similar to the String.trim() function, this one removes the leading and
+     * trailing empty/blank lines from the line list.
+     *
+     * @param lines the list of lines, which might contain empty lines
+     * @return the lines without leading or trailing blank lines.
+     */
+    // note: this is only package private, since it is used by CommentUtil. Once CommentUtil is gone, this
+    // can be private
+    static List<String> trim(List<String> lines) {
+        if (lines == null) {
+            return Collections.emptyList();
+        }
+
+        List<String> result = new ArrayList<>(lines.size());
+        List<String> tempList = new ArrayList<>();
+        boolean foundFirstNonEmptyLine = false;
+        for (String line : lines) {
+            if (StringUtils.isNoneBlank(line)) {
+                // new non-empty line: add all previous empty lines occurred before
```

Comment:
`isNoneBlank` doesn't check the line is not empty, but actually if not a single character is a whitespace. So, "this is my comment" will fail this check for having 3 whitespaces. Is this really what you expect?

Reply:
I guess, I wanted to use `isNotBlank` - isNoneBlank is the same, but for multiple lines (it accepts a vararg ... String). So, the plural here would refer to multiple lines, not to multiple characters...